concrete-core/src/specification/engines/glwe_ciphertext_decryption.rs

```diff
@@ -0,0 +1,49 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::{
+    GlweCiphertextEntity, GlweSecretKeyEntity, PlaintextVectorEntity,
+};
+
+engine_error! {
+    GlweCiphertextDecryptionError for GlweCiphertextDecryptionEngine@
+    GlweDimensionMismatch => "The ciphertext and secret key glwe dimension must be the same.",
+    PolynomialSizeMismatch => "The ciphertext and secret key polynomial size must be the same."
+}
```

Comment:
Good catch !

As a matter of fact, this is already done at compile time because of the bounds in the trait declaration:
```rust
pub trait GlweCiphertextDecryptionEngine<SecretKey, Ciphertext, PlaintextVector>:
AbstractEngine
    where
        SecretKey: GlweSecretKeyEntity,
        Ciphertext: GlweCiphertextEntity<
            Representation = SecretKey::Representation,
            KeyFlavor = SecretKey::KeyFlavor,
        >,
        PlaintextVector: PlaintextVectorEntity<Representation = SecretKey::Representation>,
{
    ...
}
```

This line for example:
```rust
       Representation = SecretKey::Representation,
```
Ensures at compile time that the `Ciphertext` type has the same representation as the `SecretKey` type.

Reply:
I close this one and the other related comments, but feel free to ask more question if something is still unclear ;)